### DISCUSSION TITLE
Implementation

### DISCUSSION THREAD

**Discussion by jfjlaros** [jfjlaros/cpp-simple-rpc#2] [Q&A]
To do:
- Generating `device.h`.
- Support for strings, vectors, tuples and objects.

**Comment by jfjlaros** [comment]
I have most basic cases working now. There is no support for strings, vectors and the like, but simple functions should work.

@slaff, perhaps you can give it a try?

I added a quick start to help you on your way.

**Comment by jfjlaros** [comment]
I also implemented a dynamic approach. This does not use a generated header file, so it should be easier to work with. It would require run time type checking though, but that should not be too hard to implement.

An example can be found [here](https://github.com/jfjlaros/cpp-simple-rpc/blob/master/examples/demo/demo.cc).

**Comment by slaff** [comment]
@jfjlaros Thanks a lot for your work. The initial version of the RPC communication in Sming based on your simpleRPC library can can be found here: https://github.com/SmingHub/Sming/pull/2305.  I have noticed one issue but I will open another topic to discuss it.

**Reply by jfjlaros** [reply]
This is only the device side of the library I guess?

**Reply by slaff** [reply]
Actually both - on the device we flash an application that is acting as the RPC server. Then on the client during link time we replace some functions with their RCP versions and the Basic_Blink can turn on and off an actual LED on the device while running all other logic on the desktop.

**Reply by jfjlaros** [reply]
Okay, and you are using the "dynamic" approach for this? I.e., without the generated header file?
